src/lib/EventListener/ContentProxyCreateDraftListener.php

```diff
@@ -46,50 +49,111 @@ public function __construct(
-            ContentProxyCreateEvent::class => 'createDraft',
+            ContentProxyCreateEvent::class => 'create',
+            ContentProxyTranslateEvent::class => 'translate',
         ];
     }
 
-    public function createDraft(ContentProxyCreateEvent $event)
+    public function create(ContentProxyCreateEvent $event): void
     {
         $isAutosaveEnabled = $this->userSettingService->getUserSetting('autosave')->value === AutosaveSetting::ENABLED_OPTION;
 
         if (!$isAutosaveEnabled) {
             return;
         }
 
-        $createContentTypeStuct = $this->contentService->newContentCreateStruct(
+        $options = $event->getOptions();
+
+        $createContentStruct = $this->contentService->newContentCreateStruct(
             $event->getContentType(),
             $event->getLanguageCode()
         );
 
         $contentDraft = $this->contentService->createContent(
-            $createContentTypeStuct,
+            $createContentStruct,
             [
                 $this->locationService->newLocationCreateStruct($event->getParentLocationId()),
             ],
             []
         );
 
-        if (!$event->getOptions()->get('isOnTheFly', false)) {
+        if ($options->get('isOnTheFly', false)) {
             $response = new RedirectResponse(
-                $this->router->generate('ezplatform.content.draft.edit', [
+                $this->router->generate('ezplatform.content_on_the_fly.edit', [
                     'contentId' => $contentDraft->id,
                     'versionNo' => $contentDraft->getVersionInfo()->versionNo,
-                    'language' => $event->getLanguageCode(),
+                    'languageCode' => $event->getLanguageCode(),
+                    'locationId' => $contentDraft->contentInfo->mainLocationId,
                 ])
             );
         } else {
             $response = new RedirectResponse(
-                $this->router->generate('ezplatform.content_on_the_fly.edit', [
+                $this->router->generate('ezplatform.content.draft.edit', [
                     'contentId' => $contentDraft->id,
                     'versionNo' => $contentDraft->getVersionInfo()->versionNo,
-                    'languageCode' => $event->getLanguageCode(),
-                    'locationId' => $contentDraft->contentInfo->mainLocationId,
+                    'language' => $event->getLanguageCode(),
                 ])
             );
         }
 
         $event->setResponse($response);
     }
+
+    public function translate(ContentProxyTranslateEvent $event): void
+    {
+        $isAutosaveEnabled = $this->userSettingService->getUserSetting('autosave')->value === AutosaveSetting::ENABLED_OPTION;
+
+        if (!$isAutosaveEnabled) {
+            return;
+        }
+
+        $content = $this->contentService->loadContent(
+            $event->getContentId(),
+            empty($event->getFromLanguageCode())
+                ? null
+                : [$event->getFromLanguageCode()]
```

Comment:
Nitpick: `empty` will be true for empty string (`""`).

```suggestion
            isset($event->getFromLanguageCode())
                ? [$event->getFromLanguageCode()]
                :  null
```

or 

```suggestion
            $event->getFromLanguageCode() === null
                ? [$event->getFromLanguageCode()]
                :  null
```